### PR TITLE
Update react wrappers project 

### LINKED ISSUE #874: Update React Wrappers project template to Create React App project
- [x] implementation
 - [ ] update [case](https://crm.infragistics.com/igCRMExtension/CHS/CaseSearch.aspx?CaseNumber=CAS-213729-R4M3X0) when this is fixed

### Description  

 - Ignite UI CLI version: 8.1.0
- Framework: React
- Project type: ig-ts

### Steps to reproduce  

Run igniteui-cli wizard using "ig" command and choose the following steps:

? Enter a name for your project: newProject
? Choose framework: React
? Choose the type of project: Ignite UI for jQuery React Wrappers
  Proj Template: empty
? Choose the theme for the project: infragistics
  Generating project structure.
v Project structure generated.
Git initialization failed. Install Git in order to automatically commit the project.

? Choose an action: Complete & Run
? Choose app host port: 3002
Build started.
Installing npm packages
Packages installed successfully
Starting project...

### Result  
 A complete log of this run can be found in:
npm ERR!     C:\Users\nalipiev\AppData\Roaming\npm-cache\_logs\2021-08-19T13_34_50_688Z-debug.log
(node:5476) UnhandledPromiseRejectionWarning: Error: Command failed: npm start -- --port=3002
    at checkExecSyncError (child_process.js:760:11)
    at Object.execSync (child_process.js:833:15)
    at Function.execSync (C:\Users\nalipiev\Desktop\newDemo@8.1.0\node_modules\@igniteui\cli-core\util\Util.js:273:36)
    at Object.<anonymous> (C:\Users\nalipiev\Desktop\newDemo@8.1.0\node_modules\igniteui-cli\lib\commands\start.js:82:41)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\nalipiev\Desktop\newDemo@8.1.0\node_modules\igniteui-cli\lib\commands\start.js:5:58)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:5476) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:5476) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

### Expected result  
The project is build and running

### Updates & Supplemental Fixes
 - [x] Using `npm run start` passing `deffaultPort` is not platform agnostic for passing a port param to react-scripts --> use **.env** file with variable **PORT=3002** instead. It will ensure the app uses it when **_"start": "react-scripts start"_** is called. This port will be used instead of the deffaultPort prop under `ignite-ui-cli.json` and deffaultPort is to be considered obsolete in the `react proj` context. The user can manually update the PORT variable to one he wishes.

- [x] Fix non-working templates as all *grid-exporting templates + Add Missing package as  SaveAs error is thrown on export.

- [x] If exporting template is added - build errors because of filename camelCase difference 
    _ext_collections**E**xtended.js/ext_collectionsextended.js
    Cannot find file: 'infragistics.ext_collectionsExtended.js' does not match the corresponding name on disk:
    '.\node_modules@infragistics\ignite-ui-full\en\js\modules\infragistics.ext_collectionsextended.js'._
The **fix is implemented** and is to be released with the next SR for the public.

 - [x] Include fix for required ignite-ui to ignite-ui-full package update based on components added:  Fallback to **@latest** when igniteui-full package version does not have an exact match for the freeOS version used. 

### Related:
 -  https://github.com/IgniteUI/igniteui-cli/issues/890           
 -  https://github.com/IgniteUI/ignite-ui/issues/2156
 -  https://github.com/IgniteUI/igniteui-cli/issues/891

### PR DESCRIPTION
**Pull request by IvayloG**
Closes #874 

Update react wrappers (ig-es6) project to use up to date Create React App project structure.

**Added:**
- Navigation component.
- .env file containing default serve port.
- Fixed not working project templates.
- Fixed using @latest **ignite-ui-full** if no exact matching upgrade version is found.
- Using barrel imports for igniteui -dependencies via igniteuiResources.js
- Other minor fixes

**TBD:**
- works with "@infragistics/ignite-ui-full": "21.1.11", however new errors with latest  "21.1.25"
#890

### CODE REVIEW

File: packages/cli/templates/react/es6/projects/empty/files/src/igniteuiResources.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Ignite UI Required Style files
+export * from "ignite-ui/css/themes/infragistics/infragistics.theme.css";
```
Comment (Lipata) [inline packages/cli/templates/react/es6/projects/empty/files/src/igniteuiResources.js:2]:
```suggestion
import "ignite-ui/css/themes/infragistics/infragistics.theme.css";
```

Even with this fix I get the following error, when running the project:

image

**Review by Lipata — CHANGES_REQUESTED** [review]

File: packages/cli/templates/react/es6/projects/empty/files/src/igniteuiResources.js (hunk 1 above)
Comment (dafo) [reply packages/cli/templates/react/es6/projects/empty/files/src/igniteuiResources.js]: I was also able to reproduce the same issue.

**Comment by Lipata** [comment]
We will wait for the fit() test and then we will merge the PR. All samples are working, except excel exporters one, for which there is a issue logged for jQuery build, to fix the exporter file name. There are also some warnings coming from the Ignite UI for jQuery, which needs refactoring there and this will not happen:

image
image

**Comment by Lipata** [comment]
#891

**Pull request merged**